edb/server/protocol/protocol.pyx

```diff
@@ -724,6 +722,59 @@ cdef class HttpProtocol:
         response.status = http.HTTPStatus.BAD_REQUEST
         response.close_connection = True
 
+    async def _handle_cors(
+        self,
+        HttpRequest request,
+        HttpResponse response,
+        *,
+        str dbname = None,
+        list allow_methods = None,
+        list allow_headers = [],
+        list expose_headers = None,
+        bint allow_credentials = False
+    ):
+        db = self.tenant.maybe_get_db(dbname=dbname) if dbname else None
+        
+        config = None
+        if db is not None:
+            if db.db_config is None:
+                await db.introspection()
+
+            config = db.db_config.get('cors_allow_origins')
+        if config is None:
+            config = self.tenant.get_sys_config().get('cors_allow_origins')
+
+        allowed_origins = config.value if config else None
+
+        if allowed_origins is None:
+            return False
+
+        origin = request.origin.decode() if request.origin else None
+        origin_allowed = origin is not None and (
+            origin in allowed_origins or '*' in allowed_origins)
+
+        if origin_allowed:
+            response.custom_headers['Access-Control-Allow-Origin'] = origin
+            if expose_headers is not None:
+                response.custom_headers['Access-Control-Expose-Headers'] = \
```

Comment:
I would suggesting using an open paren to continue the line instead of a backslash